File: src/main/java/swf/army/mil/blog/comment/Comment.java
package swf.army.mil.blog.comment;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import swf.army.mil.blog.post.Post;

import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "post_id", nullable = false)
    private Post post;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private String author;


}

File: src/main/kotlin/com/personal/blog/comment/Comment.kt
package com.personal.blog.comment

import com.personal.blog.post.Post
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import org.springframework.data.annotation.CreatedDate
import java.time.Instant

@Entity
class Comment (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @ManyToOne
    @Column(nullable = false)
    val parentPost: Post,

    val text: String,

    @CreatedDate
    @Column(nullable = false, updatable = false)
    val created: Instant = Instant.now()
)